IoTDeviceSimulator/azure-functions/d2c-simulator/index.js

```diff
@@ -208,17 +383,25 @@ module.exports = async function (context, req) {
             await client.setOptions(modelIdObject);
 
             await client.open();
-            intervalTokens.push(SetInterval(client, device));
+            intervalTokens[device.deviceId] = SetInterval(client, device);
 
             client.on('message', async function (msg) {
-              stop = true;
-              intervalTokens.forEach((value) => clearInterval(value));
-              intervalTokens = [];
-              if (req.body.enableLogging) {
-                const blockBlobClient = containerClient.getBlockBlobClient(blobName);
-                const data = JSON.stringify(jsonData);
-                // Upload data to the blob              
-                const uploadBlobResponse = await blockBlobClient.upload(data, data.length);
+              if (msg.messageId.includes("stop")) {
+                stop = true;
+                for (const key in intervalTokens) {
+                  clearInterval(intervalTokens[key]);
+                }
+                intervalTokens = [];
+                if (req.body.enableLogging) {
+                  const blockBlobClient = containerClient.getBlockBlobClient(blobName);
+                  const data = JSON.stringify(jsonData);
+                  // Upload data to the blob              
+                  const uploadBlobResponse = await blockBlobClient.upload(data, data.length);
+                }
+              }
+              else {
+                clearInterval(intervalTokens[device.deviceId]);
```

Comment:
Add a specific check for "resolve" action:
if (msg.messageId.includes("resolve")) {